### ISSUE TITLE
The loss is NaN when ET models is set "distance_influence" to none

### ISSUE THREAD

**Issue by XJTUNR** [torchmd/torchmd-net#65]
@raimis @PhilippThoelke 

Hi, authors!

I recently trained the QM9 dataset with your powerful ET model, and I found that when "distance_influence" is set to "keys" or "values" or "both", the ET model works well. However, when I set "distance_influence" to "none", the loss becomes NaN in the back propagation of the first batch.

Did you notice this phenomenon during your training? How can I fix it? I'm really curious to see how powerful your novel attention mechanism is.

Here is my training commands and results:

`CUDA_VISIBLE_DEVICES=0 python scripts/train.py --conf examples/ET-QM9.yaml --dataset QM9 --log-dir ./edge-att-output/ --dataset-root ./data`
`CUDA_VISIBLE_DEVICES=0 python scripts/train.py --conf examples/ET-QM9.yaml --dataset QM9 --log-dir ./keys-edge-att-output/ --distance-influence keys --dataset-root=./data`

They both work well.

`CUDA_VISIBLE_DEVICES=0 python scripts/train.py --conf examples/ET-QM9.yaml --dataset QM9 --log-dir ./no-edge-att-output/ --distance-influence none --dataset-root=./data`

When running the above command, I got the NaN loss:
![image](https://user-images.githubusercontent.com/52536919/160445720-17b37b4d-b256-4d8a-99a3-f9c489873a4c.png)

All experiments are running on a single V100 GPU.

I would be very grateful for your help! Thanks！

**Comment by PhilippThoelke** [comment]
Hi, thanks for reporting that! I'm able to reproduce this but I'm not 100% sure why this occurs. PyTorch's anomaly detection tells me that the NaN's are caused by the backwards pass of the layer normalization in the ET's attention mechanism: https://github.com/torchmd/torchmd-net/blob/83e6b8fca6f2a65689fe241cf7925bf3fa64b7dd/torchmdnet/models/torchmd_et.py#L265

When `distance_influence` is set to `"none"`, `x` seems to be exploding throughout the layers of the model. When `distance_influence` is something else, `x` is scaled down in later steps of the attention mechanism, which keeps it from exploding. I currently don't have a fix for this as we are only using `distance_influence=both`, which produces the best results. The argument was used during earlier stages of the development where it didn't break a different version of the architecture.
